rest/model/pod.go

```diff
@@ -149,6 +156,41 @@ func (p *APIPod) ToService() (*pod.Pod, error) {
 	}, nil
 }
 
+// APIPodType represents a pod's type.
+type APIPodType string
+
+const (
+	PodTypeAgent APIPodType = "agent"
+)
+
+// BuildFromService converts a service-layer pod type into a REST API pod type.
+func (t *APIPodType) BuildFromService(pt pod.Type) error {
+	var converted APIPodType
+	switch pt {
+	case pod.TypeAgent:
+		converted = PodTypeAgent
+	default:
+		return errors.Errorf("unrecognized pod type '%s'", pt)
+	}
+	*t = converted
+	return nil
+}
+
+// ToService converts a REST API pod type into a service-layer pod type.
+func (t *APIPodType) ToService() (*pod.Type, error) {
+	if t == nil {
```

Comment:
Do we need to check for `nil` here? It seems like panic behavior.

Reply:
I waffled on this when I wrote it because there's some remote possibility that the type could be optional during API <-> DB conversion, so we would have to handle `*APIPodType` at some point in the future. There could be an argument that the `APIPod`'s `Type` field should be `*APIPodType` for consistency with other REST models, which always make all fields optional. However, given that it's not an optional field for now and assuming the DB will always populate it (this should be true), I changed it to be a non-pointer so the receiver is now `t APIPodType`.

On the topic of panics, this doesn't panic because you can call `ToService` even if `t` is a nil pointer. I'm pretty sure the virtual method table for a type should still be valid even on a nil pointer. Ex:
```go
import (
	"fmt"
)

type foo struct{}

func (f *foo) bar() {
	fmt.Println("I don't panic")
}

func main() {
	var f *foo
	f.bar()
}
```